crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,264 @@
+//! Shared sequencer client
+
+#![deny(clippy::arithmetic_side_effects)]
+#![deny(clippy::cast_possible_truncation)]
+#![deny(unused_crate_dependencies)]
+#![deny(missing_docs)]
+
+use anyhow::anyhow;
+use cosmrs::{
+    tendermint::chain::Id,
+    tx::{
+        self,
+        Fee,
+        MessageExt,
+        SignDoc,
+        SignerInfo,
+    },
+    AccountId,
+    Coin,
+    Denom,
+};
+use error::PostBlobError;
+use fuel_sequencer_proto::protos::fuelsequencer::sequencing::v1::MsgPostBlob;
+use http_api::{
+    AccountMetadata,
+    TopicInfo,
+};
+use ports::Signer;
+use prost::Message;
+use tendermint_rpc::Client as _;
+
+// Re-exports
+pub use config::{
+    Config,
+    Endpoints,
+};
+pub use prost::bytes::Bytes;
+
+mod config;
+mod error;
+mod http_api;
+pub mod ports;
+pub mod service;
+
+/// Shared sequencer client
+pub struct Client {
+    endpoints: Endpoints,
+    topic: [u8; 32],
+    chain_id: Id,
+    gas_price: u128,
+    coin_denom: Denom,
+    account_prefix: String,
+}
+
+impl Client {
+    /// Create a new shared sequencer client from config.
+    pub async fn new(endpoints: Endpoints, topic: [u8; 32]) -> anyhow::Result<Self> {
```

Comment:
i think it should also accept the expected client_id so that we don't misconfigure with the wrong network/chain

Reply:
What is the `client_id`? Do you mean `chain_id`? I want to keep the current implementation as simple, as possible. 

Plus, if the RPC is malicious, it can lie about its `chain_id`

Reply:
sorry, `chain_id` :)